### DISCUSSION TITLE
Error! in new Wemos d1 mini !!!

### DISCUSSION THREAD

**Discussion by TrDA-hab** [arendst/Tasmota#11615] [Show and tell]
Yesterday I received two pieces of new wemos d1 mini. I bought them from this store:
https://aliexpress.ru/item/4001291931302.html

I flashed both wemos and got this problem for every new wemos:

![1](https://user-images.githubusercontent.com/56403720/113756053-a699b580-9719-11eb-921d-bb88f51bedab.jpg)
... after a very long time (every time the connection time is random)
![11](https://user-images.githubusercontent.com/56403720/113756590-3fc8cc00-971a-11eb-89d7-fd4735b7585c.jpg)

After rebooting my router, wemos got an IP address. And immediately wemos froze and could no longer get the IP dynamic address (I rebooted the router many times):
![2](https://user-images.githubusercontent.com/56403720/113756090-aef1f080-9719-11eb-92f0-8e0e7aa1cb5c.jpg)

I tried to fix the problem:
- reboot the router many times.
- to flash through PlatformIO and through tasmotizer and through a third-party flasher.
- tried to use static and dynamic IP addresses.
- tried to use my phone as a router.
- I tried all four (!) firmware modes.
- i tried installing different versions 9.3.X and 9.0.X and 8.5.X
- I even replaced the memory chip in one of the new wemos.
![20210406_160131](https://user-images.githubusercontent.com/56403720/113757163-f75dde00-971a-11eb-8f75-8007ab23c440.jpg)
![20210406_161141](https://user-images.githubusercontent.com/56403720/113757184-fdec5580-971a-11eb-8054-c121c8c38e7e.jpg)
![20210406_161651](https://user-images.githubusercontent.com/56403720/113757205-03e23680-971b-11eb-9cb9-c482dc4e1244.jpg)

I have flashed firmware from another manufacturer and received the message “RTC memory data should not be used”.![4](https://user-images.githubusercontent.com/56403720/113756114-b4e7d180-9719-11eb-9988-0cf4dc8c6b6c.jpg)

I have no more good ideas. Old wemos are flashed and work without problems - I checked this several times today trying to reproduce this error.

UPD#1.
I tried to fix the problem (optional):
- supply power to Wemos via pin 3.3V from three different power sources.
- add the capacitor (s) to the 3.3V power line.

Wemos began to load (every second time), but it works very slowly - it responds to commands in 7-10 seconds.

Avoid buying Wemos with this memory chip!

![20210406_153452](https://user-images.githubusercontent.com/56403720/113908859-afed5580-97df-11eb-807a-0b6d7fe08b6d.jpg)

UPD#2
Old and New ESP chips (no difference = same).
![20210406_165850](https://user-images.githubusercontent.com/56403720/113910436-82091080-97e1-11eb-9305-58ee82ef5c93.jpg)
![20210406_165952](https://user-images.githubusercontent.com/56403720/113910461-87665b00-97e1-11eb-8423-16da1774a0f5.jpg)

***Best regards,
TrDA***

**Comment by swampdogmash** [comment]
Is this maybe just a case of ensuring the correct flash mode settings when flashing? did you use DIO for the wemos?

**Reply by TrDA-hab** [reply]
> Is this maybe just a case of ensuring the correct flash mode settings when flashing? did you use DIO for the wemos?

I tried all four (!) firmware modes.

**Comment by Jason2866** [comment]
Have you replaced the onboard 3.3V vreg? The issues you have are typical for bad powering the ESP.

**Reply by TrDA-hab** [reply]
No, I haven't tried replacing the 3.3V regulator (it doesn't make sense). I applied power to the 3.3V pin ... and nothing has changed.

**Comment by nonix** [comment]
I've just checked my, I have similar variant in Pro version. Flashed the development branch (direct download) using Tasmotizer v1.2 with options: Wemos (self reseting) and Erase. Than, uploaded config to my match my WiFi.
Seems all fine.

The only thing I have noticed, the micro USB connector is a bit clumsy. If so, try to power up using a good source 3.3V direct on 3V pin.

Hope this helps,
 N.


```
 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 3584, room 16
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

00:00:00.049 CFG: Loaded from flash at FA, Count 10
00:00:00.054 QPC: Count 1
00:00:00.056 CFG: CR 354/699, Busy 0
00:00:00.060 SRC: Restart
00:00:00.062 Project tasmota Tasmota Version 9.3.1.2(tasmota)-2_7_4_9(2021-04-06T13:25:05)
00:00:00.213 WIF: Attempting connection...
00:00:00.549 WIF: Connecting to AP1 0325125441 Channel 1 BSSId XXXXXXXXXXX in mode 11n as tasmota_E3B734-5940...
00:00:01.753 WIF: Connected
00:00:02.003 HTP: Web server active on tasmota_E3B734-5940 with IP address 192.168.13.129
00:00:02.443 RTC: UTC 2021-04-06T19:26:06, DST 2021-03-28T02:00:00, STD 2021-10-31T03:00:00
21:26:07.006 RSL: tele/tasmota_E3B734/INFO1 = {"Module":"Generic","Version":"9.3.1.2(tasmota)","FallbackTopic":"cmnd/DVES_E3B734_fb/","GroupTopic":"cmnd/tasmotas/"}
21:26:07.011 RSL: tele/tasmota_E3B734/INFO2 = {"WebServerMode":"Admin","Hostname":"tasmota_E3B734-5940","IPAddress":"192.168.13.129"}
21:26:07.023 RSL: tele/tasmota_E3B734/INFO3 = {"RestartReason":"Software/System restart"}
21:26:10.467 QPC: Reset
21:26:11.377 HTP: Main Menu
21:26:12.402 APP: Boot Count 4
21:26:12.581 CFG: Saved to flash at F9, Count 11, Bytes 4096

```

![WemosD1_pro](https://user-images.githubusercontent.com/1888777/113767100-8ae4de00-971e-11eb-85ac-1593558b1c4b.jpg)

**Reply by nonix** [reply]
Hi did you get any further? It seems there is something odd with this "new" clones. I can confirm the resetting / rebooting as well ... :-(

**Reply by TrDA-hab** [reply]
> Hi did you get any further? It seems there is something odd with this "new" clones. I can confirm the resetting / rebooting as well ... :-(

No, I didn't succeed.

**Reply by nonix** [reply]
In your config, do you use ADC? I think it does have something w. ADC. My setup is a simple thermometer w. display both components are I2C. However, I have tried to connect LDR to ADC, in next few seconds (perhaps a minute) it resets and restore defaults.
N.

**Reply by Jason2866** [reply]
Trash it. You cant solve bad hardware (design).

**Reply by nonix** [reply]
@Jason2866 I do disagree (at least in this case) It seems there is bug in the tasmota when "ADC Button" is used. I can reproduce the same issue over and over with the HW board mentioned above as well as with ESP12F based Wemos mini.
N.

**Reply by Jason2866** [reply]
The board is failing in a very early phase. This can not be related to anything in Tasmota.
So the conclusion.
No one was talking about ADC usage.

**Comment by nonix** [comment]
'm not sure how relevant this will be to your issue, but here is the issue I was dealing with just yesterday. 
When I have powered the D1 mini direct via rail, the RST was holding low as if the RTS is low and DTR is high (or high enough to open the gate)

The solution for me was to solder a 10uF cap parallel with D2 diode, positive on +5V. The idea is to give a short pulse to SIL (via REGIN) to "program" the outputs so that the RTS/DTR stays low on both and result in keeping the ESP RESET high.
[D1 mini schematic](https://escapequotes.net/wp-content/uploads/2017/06/esp8266-wemos-d1-mini-pro-schematics.png)   (schematic is not a 100% match but close enough) 

Hope this could help someone ...
N.

**Reply by barbudor** [reply]
Power via rail means powering 5V on the edge connector (not on the USB connector) ?
In that case, I can't understand what a capacitor across D2 would do if USB is not connected.

**Reply by nonix** [reply]
Well, that was a mystery to me too, but it seems there is an idea for power saving. Basically you need the SIL2104 only when USB is plugged in. When powering via rail, the SIL2104 is in "sleep" mode (hence the REGIN via VBUS) there is D2 diode which will keep REGIN low. However, when the SIL is in "sleep" the RTS/DTR is in unknown state or floating (I don't have a proof, just a speculation/assumption). That spark up an idea, to wake-up the SIL just for a brief moment (while cap is charging) to settle the RTS/DTR (keeping the ESP RESET at high via transistor inverter and pull-up). Well, all of this is just my hobby knowledge and I could be well wrong, but the 10uF cap did the trick in my case.
Cheers,
N.
Note, this seems to be the case only for Wemos D1 mini R3 pro. The Wemos D1 mini R2 (with ESP12F chip)  seems to be working well without this mod.

**Reply by jadomingo92** [reply]
In my case you are right @nonix, the diode you mentioned seems to be the problem.
I had a bunch of wemos that worked with USB but not when powered by 5V pin (with usb unplugged), the only way the 5V pin power worked was when the USB had been plugged before. It seems the SIL chip (USB converter) is only powered when USB is used but it is connected to some mosfets that handle the RST pin in the ESP.
The diode you mentioned is what prevents the chip to be powered by the 5V pin and I think it is meant to prevent damage to the USB port in the PC when powered through both the USB cable and the 5V pin. I removed the diode and connected the pads and it works fine, but I think now its unsafe to connect the USB while the 5V pin is powered. Maybe some pullups/pulldowns are missing in the mosfet circuit in this new modules because I think when the SIL is unpowered those pins connected to the mosfets are left floating.
The small six pin IC is the double mosfet, I marked the diode I removed AND shorted. The segment of the SIL ship I marked is the one that only gets power from USB when the diode is in place. Hope this helps.
![image](https://user-images.githubusercontent.com/5928204/152886238-a54c3732-fc08-43a5-84e4-56734fff4d3b.png)
![image](https://user-images.githubusercontent.com/5928204/152886308-f7229182-1ad4-4eb2-a7ac-3b15b3df6f59.png)
